neurodamus/ngv.py

```diff
@@ -11,65 +10,114 @@
 from .connection import Connection
 from .connection_manager import ConnectionManagerBase
 from .core import MPI, EngineBase, NeuronWrapper as Nd
-from .core.configuration import GlobalConfig, LogLevel
+from .core.configuration import ConfigurationError, GlobalConfig, LogLevel
 from .io.sonata_config import ConnectionTypes
 from .io.synapse_reader import SonataReader, SynapseParameters
 from .metype import BaseCell
 from .morphio_wrapper import MorphIOWrapper
-from .utils.logging import log_verbose
-from .utils.pyutils import append_recarray, bin_search
+from .utils.pyutils import append_recarray
 
 
 class Astrocyte(BaseCell):
-    __slots__ = ("_nseg_warning", "section_names", "sections_glut", "soma_glut")
+    __slots__ = ("_gluts", "glut_soma", "section_names")
 
     def __init__(self, gid, meinfos, circuit_conf):
-        """Initialize an Astrocyte cell"""
+        """Initialize an Astrocyte cell:
+
+        - create the cell from Cell.hoc
+        - add the morphology
+        - resize sections if necessary
+        - add cadifus, GlutReceive (they will be connected later)
+        - record section_names for creating connections later
+        """
         super().__init__(gid, meinfos, None)
 
-        # Compose the path to the morphology file
-        morph_file = (
+        # Create the cell
+        self._cellref = Nd.Cell(gid)  # cell instantiated with Cell.hoc
```

Comment:
The morphology loading in Cell.hoc has been made more robust compared to when first working with earlier ngv development. Maybe a future PR can see if there are still changes necessary in order to simply call Cell( gid, morphfile)?

Reply:
I did not know that was a possibility. I can open an issue

Reply:
#286